.github/workflows/release.yml

```
@@ -7,14 +7,38 @@ on:
   push:
     tags:
       - 'v*'
+  pull_request:
+    types:
+      # explicitly specifying the event that triggers the workflow to provide clarity
```

I think what we actually want here is 
```suggestion
  push:
    branches: ["main"]
```

Since main is branch_protected, we will only get a push to main when a PR merges.
I believe pull_request.closed is when someone closes a pull_request, not merge, could be wrong though.